cm2350/e200z7.py

```diff
@@ -544,8 +436,60 @@ def debug_client_detached(self):
         Handles the condition when a gdb client detaches. For now we will halt
         the emulator when this occurs.
         '''
-        self.queueException(intc_exc.GdbClientDetachEvent())
-        self.resume_exec()
+        # if va is already a breakpoint, don't do it!!
+        if va in self._bpdata:
+            return -1   # raise exception
+
+        brkbytes = self.arch.archGetBreakInstr()
```

Comment:
Will this be the same for VLE and BOOKE?

Reply:
The `archGetBreakInstr()` function is commented in the PowerPC architecture module. there are 3 different break instructions that could be returned depending on context and there is no good way to distinguish between which one gets returned `vivisect/envi/archs/ppc/__init__.py`. It would require knowledge of the address the breakpoint is being placed, and if the memory address is in a VLE code section the available space available for the break instruction to be inserted.
```
    def archGetBreakInstr(self):
        # BOOKE:        dnh     4C00018C
        # VLE 32bit:    e_dnh   7C0000C2
        # VLE 16bit:    se_dnh  000F
        return b'\x4C\x00\x01\x8C'
```